src/main/scala/com.snowplowanalytics.snowplow/scalatracker/Tracker.scala

```diff
@@ -118,14 +148,89 @@ class Tracker(emitters: Seq[TEmitter], namespace: String, appId: String, encodeB
   }
 
   /**
+   * Track a Snowplow structured event
+   *
+   * @param category event category mapped to se_ca
+   * @param action event itself mapped to se_ac
+   * @param label optional object label mapped to se_la
+   * @param property optional event/object property mapped to se_pr
+   * @param value optional object value mapped to se_va
+   * @param contexts list of additional contexts
+   * @param timestamp optional user-provided timestamp (ms) for the event
+   * @return the tracker instance
+   */
+  def trackStructEvent(
+    category: String,
+    action: String,
+    label: Option[String] = None,
+    property: Option[String] = None,
+    value: Option[Double] = None,
+    contexts: Seq[SelfDescribingJson] = Nil,
+    timestamp: Option[Long] = None): Tracker = {
+
+    val payload = new Payload()
+
+    payload.add("e", "se")
+    payload.add("se_ca", category)
+    payload.add("se_ac", action)
+    payload.add("se_la", label)
+    payload.add("se_pr", property)
+    payload.add("se_va", value.map(_.toString))
+
+    track(completePayload(payload, contexts, timestamp))
+
+    this
+  }
+
+  /**
+   * Record view of web page
+   *
+   * @param pageUrl viewed URL
+   * @param pageTitle page's title
+   * @param referrer referrer URL
+   * @param contexts list of additional contexts
+   * @param timestamp optional user-provided timestamp (ms) for the event
+   * @return the tracker instance
+   */
+  def trackPageView(
+    pageUrl: String,
+    pageTitle: Option[String] = None,
+    referrer: Option[String] = None,
+    contexts: Seq[SelfDescribingJson] = Nil,
+    timestamp: Option[Long] = None): Tracker = {
+
+    val payload = new Payload()
+
+    payload.add("e", "pv")
+    payload.add("url", pageUrl)
+    payload.add("page", pageTitle)
+    payload.add("refr", referrer)
+
+    track(completePayload(payload, contexts, timestamp))
+
+    this
+  }
+
+  /**
    * Set the Subject for the tracker
    * The subject's configuration will be attached to every event
    *
-   * @param subject
+   * @param subject user which the Tracker will track
    * @return The tracker instance
    */
   def setSubject(subject: Subject): Tracker = {
     this.subject = subject
     this
   }
+
+
+  /**
+   * Adds EC2 context to each sent event
+   * This will also make tracker to wait for complete AWS EC2 request
+   * before send all events
+   */
+  def enableEc2Context: Unit = {
+    attachEc2Context = true
+    ec2Context.completeWith(Ec2Metadata.getInstanceContextFuture)
+  }
```

Comment:
I think this should be `def enableEc2Context()` so that it can be called with brackets. Calling a nullary method with brackets gives a visual cue that the method is side-effecting.